Presentation.Web/app/utility/kendoGridLauncher.ts

```diff
@@ -838,16 +836,57 @@ module Kitos.Utility.KendoGrid {
                                     dataTextField: "text",
                                     dataValueField: "id",
                                     optionLabel: entry.title,
+                                    enabled: true,
                                     change: e => {
-                                        var selectedId = e.sender.value();
-                                        const newSelection = entry.dropDownConfiguration.availableOptions.filter(x => x.id === selectedId);
-                                        entry.dropDownConfiguration.selectedOptionChanged(newSelection.length > 0 ? newSelection[0] : null);
+                                        this.triggerSelectedOptionChanged(e, entry);
                                         this.saveGridOptions();
                                     }
                                 }
                             }
                         };
                         break;
+                    case KendoToolbarImplementation.CustomTemplateDropDownList:
+
+                        const dropdown = {
+                            name: entry.id,
+                            text: entry.title,
+                            template: `<select data-element-type='${entry.id}DropDownList' id='${entry.id}' class='${Helpers.KendoToolbarCustomizationHelper.getPositionClass(entry.position)} ${Helpers.KendoToolbarCustomizationHelper.getMargins(entry.margins)}' kendo-drop-down-list="kendoVm.${entry.id}.list" k-options="kendoVm.${entry.id}.getOptions()"></select>`
+                        };
+
+                        if (entry.id === Constants.CustomFilterDropdown.Id) {
```

Comment:
what?.... it reordering the toolbar.... in the builder?
Why not where the fixed toolbar items are layed out?